### DISCUSSION TITLE
Run action only when a PR is merged.

### DISCUSSION THREAD

**Discussion by CalvinRodo** [community/community#26449] [Actions]
<p>I have the following workflow file: <br>
<a href="https://github.com/cds-snc/c19-benefits-node/blob/master/.github/workflows/build-deploy-dev.yml" rel="nofollow noopener">https://github.com/cds-snc/c19-benefits-node/blob/master/.github/workflows/build-deploy-dev.yml</a></p>
<p>I only want it to run when a PR get’s merged to the master branch, however it’s also running when I open a PR from the master branch to another branch.</p>
<p>What’s the proper even to use for this situation, or can someone explain why opening a PR is considered a <code>push</code> on a branch?</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/calvinrodo">@calvinrodo</a> ,</p>
<p>I think you confuse **<a href="https://help.github.com/en/actions/reference/events-that-trigger-workflows#pull-request-event-pull_request" rel="nofollow noopener">Pull request event</a> **with <a href="https://help.github.com/en/actions/reference/events-that-trigger-workflows#push-event-push" rel="nofollow noopener"><strong>Push event</strong></a> .</p>
<p>Few points may be able to help you understand:</p>
<p><strong>1)</strong> If you want the workflow runs only for PR, you should set it runs on <strong>Pull request event</strong> , not <strong>Push event</strong>.</p>
<pre><code>**on: pull\_request**
</code></pre>
<p><strong>2)</strong> Once a PR is merged, this PR also is closed automatically. If you want the workflow runs only when the PR is merged to <strong>master</strong> branch, you can set branches filter and activity types for the <strong>Pull request event</strong>**.**</p>
<pre><code>**on: pull\_request: types: [closed] branches: - master**
</code></pre>
<p><strong>3)</strong> When using the activity type  <strong>pull_request.closed</strong> , once the PR is closed the workflow will be triggered, regardless of whether the PR is merged or not. If you want the workflow runs only when the PR is merged, you can add an <a href="https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idif" rel="nofollow noopener"><strong>if conditional</strong></a> on each job to skip all the jobs in the workflow when the PR is closed without merged. Here you need using the property " <strong>github.event.pull_request.merged"</strong> of the <a href="https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context" rel="nofollow noopener"><strong>github context</strong></a> into the  <strong>if conditional</strong>.</p>
<pre><code>**if: github.event.pull\_request.merged == 'true'**
</code></pre>
<p><strong>4)</strong> Summarizing all the above points, you can configure your workflow like the following:</p>
<pre><code>**on: pull\_request: types: [closed] branches: - master jobs: job1: name: Job 1 if: github.event.pull\_request.merged == 'true' job2: name: Job 2 if: github.event.pull\_request.merged == 'true' . . . jobN: name: Job N if: github.event.pull\_request.merged == 'true'**
</code></pre>
